### PR TITLE
Use bazel version 3.1.0, update variants to 3.4.1

### PR DESCRIPTION
**Pull request by fejta**
Once these images are pushed I should be able to bump to 3.4.1.

### CODE REVIEW

**Comment by chaodaiG** [comment]
Please also update https://github.com/kubernetes/test-infra/blob/7e74b33534f950170801fb3ce015c1e96c975471/prow/test/integration/test.sh#L38 and https://github.com/kubernetes/test-infra/blob/7e74b33534f950170801fb3ce015c1e96c975471/prow/test/integration/test.sh#L42

File: images/bazel/variants.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ variants:
     OLD_VERSION: 0.29.1
   test-infra:
     CONFIG: test-infra # Used by test-infra repo
+    OLD_VERSION: 3.4.1
```
Comment (chaodaiG) [inline images/bazel/variants.yaml:12]: Should 3.4.1 be `NEW_VERSION`?

Comment (fejta) [reply images/bazel/variants.yaml]: Sure, although doesn't really matter

**Comment by fejta** [comment]
> Please also update

Done

**Review by michelle192837 — APPROVED** [review]

**Review by chaodaiG — APPROVED** [review]

**Comment by chaodaiG** [comment]
RBE disk quota increased, try again
/test pull-test-infra-integration

**Comment by chaodaiG** [comment]
`pull-test-infra-integration` is blocked by #20628 , which is what this PR is trying to fix. Overriding it to make this fix go through.
/override pull-test-infra-integration

**Comment by chaodaiG** [comment]
Tide doesn't seem to handle the override very well in this scenario, overriding all to see if it'll work
/override pull-test-infra-verify-file-perms pull-test-infra-yamllint

**Comment by chaodaiG** [comment]
Tide doesn't seem to be merging this PR even with all contextes overridden, manually merging the PR

**Comment by cjwagner** [comment]
Tide works with the override plugin and is behaving as expected here. Tide prefers batches over serial merges so it won't merge an individual PR while a batch is being tested to avoid invalidating the batch. Since the bazel jobs are taking 2h15mins to complete the batches are blocking the merge queue for that amount of time.
https://prow.k8s.io/?repo=kubernetes%2Ftest-infra&type=batch

**Pull request merged**